app/services/sufia/workflow/workflow_by_admin_set_strategy.rb

```diff
@@ -0,0 +1,16 @@
+module Sufia
+  module Workflow
+    class WorkflowByAdminSetStrategy
+      attr_reader :admin_set_id
+      def initialize(work, attributes)
+        @work = work
+        @admin_set_id = attributes[:admin_set_id] unless attributes[:admin_set_id].nil?
```

Comment:
I suspect you should just set the ivar here and not do the `nil` check yet, else when you try to reference the ivar/attr in your other method, it may be undefined.